### DISCUSSION TITLE
Add back the option to expand sidebar on hover

### DISCUSSION THREAD

**Discussion by opticSquid** [zen-browser/desktop#3696] [Ideas]
Previously there was an option to hover over the side tab bar to expand it. But now, when I'm using "Collapsed Toolbar", I don't see that option to re-enable it. Without this expand on hover it is very unintuitive to close a tab without using keyboard shortcuts.

https://github.com/user-attachments/assets/a65e3b55-87b9-4462-a9aa-ffca83707237

**Comment by Gorgaer** [comment]
Try Sidebar Expand on Hover mod. It works pretty well.

**Reply by opticSquid** [reply]
Ok I will try. But still it should be a core feature only. We should not require an extension for that

**Reply by Gorgaer** [reply]
I completely agree with you. This is a core feature and it should not have been removed.

**Reply by thothi** [reply]
Unfortunately it's broken in the latest version (1.0.2-b.2) The tooltext still appears where it should, but the sidebar doesn't expand.
Strangely enough when you enable compact mode and hide the tab bar, then it works. I'm not a fan of hiding the tab bar though, so I'm hoping for a fix

![image](https://github.com/user-attachments/assets/ccc91e45-3c1a-4881-a96a-712201bd7f97)

**Reply by navdr7** [reply]
Try Twilight release, t.2 still work well and stable, and already updated to firefox 133.0.3 without need to Update to newest beta.

**Reply by pota2dev** [reply]
> Try Sidebar Expand on Hover mod. It works pretty well.

Thank you so much. I didn't know there is a mod for this.

This should be a core feature. Microsoft Edge and Brave handles collapsing vertical tabs on hover. This is actually proper way to collapsing vertical tabs.

**Comment by johnsonjamees** [comment]
Seconding this. This should be a core feature.

**Comment by johnsonjamees** [comment]
Well, it looks like he won't budge https://github.com/zen-browser/desktop/issues/3522#issuecomment-2543568420

**Comment by anasmohammed361** [comment]
this was one of the core features that made me to stick with zen , if that has been removed , then there is no option for me rather than switching browser , i think the community must also take a hand in the development and maintainability of this wonderful feature in zen rather than expecting everything from author.

**Reply by anasmohammed361** [reply]
i switched to the older version of zen `1.0.1.a.22` which still supports the expand on hover

**Comment by navdr7** [comment]
I lose the same...
Zen ask to Update per -+2 days, and some usefull features of zen then gone...? Then come some other "for what?" features like popup url bar etc... 
For me what make Zen choosen was how it feels like floorp's thing but more stable, lighter & faster. 
I found other forks just too late to update, but zen offer like i have not yet updated it before :D
And i guess next Update (again) will come very very soon...

**Comment by mr-cheffy** [comment]
Guys this isnt an airport, you dont need to say you are leaving. 

The decision of removing expand on hover was due to the sheer amount of afford needed to be taken in order to be maintained through the releases, therefor, it may be reconsidered on stable releases.

I understand some of you may be disappointed or even angry at this decision, but talk is cheap. Either go to patreon or send patches. Thanks

**Comment by apotenza92** [comment]
Trying to revive interest back in this feature. 
[This mod](https://github.com/Uiniel/zen-mods/tree/main/sidebar-expand-on-hover) has been good thus far, just has a few minor bugs in the newer versions of Zen.
Would love to see this integrated. I'd pay 50 USD to someone to put this in.

**Reply by V0LDY** [reply]
I switched to Floorp on my laptop just because it has that feature native

**Comment by vraravam** [comment]
This is now coming natively in FF (latest nightly: 138.0a1). I hope this continues into the FF stable releases, and Zen adopts the same!

**Reply by apotenza92** [reply]
Does Zen use the native implementation of Firefox's new sidebar? Maybe it should?

**Comment by pota2dev** [comment]
There used to be a mod but now it's broken. I've switched to Floorp for this reason.

**Comment by apotenza92** [comment]
I maintain my offer of 50 USD for someone to either implement the new native Firefox sidebar with Zen customisations (which may be a more stable solution moving forward but will be a lot of work), or to implement expanding sidebar as a native feature.

I think we should replace the 'Collapsed toolbar' appearance option entirely. From a UI point of view it would make more sense that if a user picks 'Multiple toolbars' there be two check boxes that appear: 'Collapsed toolbar', and if a user enables that, an additional checkbox for 'Auto expand collapsed toolbar'.

It would be amazing if there were some equivalent option for the Single toolbar option but I think that would require a whole lot more UI work to have a good looking 'collapsed' single toolbar.

**Reply by apotenza92** [reply]
@mauro-balades my talk is not (as) cheap. Any interest?

**Comment by FynnND** [comment]
Still waiting for that feature. I tried multiple mods and all of them are either completely broken

**Comment by opticSquid** [comment]
I think the maintainers have made it pretty clear that our beloved feature is not going to come back. So, there is no point keeping this issue open. If you want I would say native firefox has it now. May be give it a try.
~ Thanks everyone

**Comment by akshever** [comment]
This sucks

**Comment by psychananaz** [comment]
as always, the maintainers of zen refusing to change the thing nobody but them want.
